src/validators/validators-functions.js

```diff
@@ -0,0 +1,50 @@
+import { isFunction, isPlainObject } from '../utils/type-inference';
+import { castArray } from '../utils/array';
+
+const getValidators = (validator) => isFunction(validator) ? validator() : validator;
+
+/**
+ * check if a value is valid or not
+ * @param  {object} validators -  a config object
+ * @param  {any} value
+ * @param  {key} string
+ * @returns boolean - using the validators to check if the value is a valid value or not
+ */
+export const isValueValid = (validators, value, key) => {
+  const validatorsArray = castArray(validators);
+  const isValid = validatorsArray.some((validator) => getValidators(validator).value(value));
```

Comment:
This should be validatorsArray.every instead of .some right?

Reply:
It's 'or' not an 'and' ... if at list one is valid it defined as valid

Reply:
for example, a value that can be a number or and string